src/main.rs

```diff
@@ -0,0 +1,295 @@
+use clap::{App, Arg, SubCommand};
+use serde::{Deserialize, Serialize};
+use std::error::Error;
+
+static PARENT_FOLDER: &str = ".graft.lock";
+static TMP_FOLDER: &str = "tmp";
+static CONFIG_FILE: &str = "graft.txt";
+static LOCK_FILE: &str = "conflict.lock";
+
+#[derive(Serialize, Deserialize, Debug)]
+struct Repo {
+    upstream_path: String,
```

Comment:
Was wondering if we should make this a map. Currently the path to a tar file can mention the exact version number, and if we rely on extracting the folder name from file name by stripping .tar.gz part, then it may include version info, so we will have to write more detailed parser. It would be better if we simply had:

```json
{
    "django": "https://djangoproject.com/django-3.1-beta.tgz"
}
```

Though we need things in order, so if same file is present in multiple tar files, if we always apply in same order we always get the same output, so map may not be great idea.